src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -486,5 +479,21 @@ internal IAsyncResult BaseBeginWrite(byte[] buffer, int offset, int count, Async
             => base.BeginWrite(buffer, offset, count, callback, state);
 
         internal void BaseEndWrite(IAsyncResult asyncResult) => base.EndWrite(asyncResult);
+
+        internal static bool IsIoRelatedException(Exception e) =>
```

Comment:
it could, but then I would either have to duplicate it for `Windows` and `Unix` `FileStreamHelpers` or make `FileStreamHelpers` a partial type and have some cross-platform logic in `FileStreamHelpers.cs`. I'll try to do that in the upcoming refactoring PR.